lib/extensions.js

```diff
@@ -181,9 +181,11 @@ function getBinaryName() {
   } else if (pkg.nodeSassConfig && pkg.nodeSassConfig.binaryName) {
     binaryName = pkg.nodeSassConfig.binaryName;
   } else {
+    var jsEngine = process.jsEngine ? '-' + process.jsEngine : '';
     binaryName = [process.platform, '-',
                   process.arch, '-',
-                  process.versions.modules].join('');
+                  process.versions.modules,
+                  jsEngine].join('');
```

Comment:
Not a manditory change, I was just thinking this
```js
     binaryName = [process.platform,
                   process.arch,
                  process.versions.modules].join('-');
     binaryName += process.jsEngine ? '-' + process.jsEngine : '';
```
But maybe that isn't any more readable :)